charmhelpers/contrib/openstack/templates/grafana-haproxy-dashboard.json.j2

```diff
@@ -0,0 +1,1157 @@
+{
```

Comment:
Please have a look to [this other dashboard](https://gist.github.com/auria/ee63b3d988a10273e9cc156c26a58fb6) that I have created based on yours. I have added a "Summary" section with a bit of everything, and instant metrics... and then other sections "Frontends", "Backends" (instead of "Counters"), and the "Responses" and "Servers" that could be extended with more metrics.

I think the title should always be "[juju] HAProxy <application>" (where <application> may be "keystone" or others). The version should be kept on the `version` attribute at the end of the json file, which gets increased.

Other dashboards imported in Grafana via Juju relations have names like "[juju] Telegraf Host Metrics" and such, that's why we should keep "[juju] ..." for consistency. Besides, we want to dashboards to get updated in a transparent way, so having different dashboard titles like "haproxy 1", "haproxy 2", etc. makes explicit something that a user doesn't need to know?

Reply:
Thanks. You are right, the name should not contain "[juju-keystone]" but only "[juju]". 
As for the version, it must be in the name, as the version in json is automatically increased. e.g. `version=1` and the `grafana-relation-joined` hook is trigger one more time -> `version=2`. This will cause another charm to overwrite the current dashboard to an older version.

Reply:
I went through [charm-grafana](https://git.launchpad.net/charm-grafana), and I found that function [1] is used to create dashboard title. I can't find a way to prevent the name from being in the form "[juju-keystone] ...".

---
[1]: https://git.launchpad.net/charm-grafana/tree/src/lib/charms/layer/grafana.py#n182